docs/src/examples/ode/freefall.py

```diff
@@ -0,0 +1,145 @@
+
+# Experiments description:
+#
+# Experiments 1 and 2 have the same time of falling 't_end1', while
+# experiments 3, 4, 5 and 6 have the time 't_end2'.
+# On the other hand experiments 1 and 3 don't use the 'onroot',
+# experiments 2 and 4 do and compute until the time t_end is reached
+# (function onroot_va()). Experiment 5 stops after the first interruption
+# (function onroot_vb()) occurs, whereas experiment 6 stops after the
+# first interruption  at time t>28 (s) (function onroot_vc()).
+# Otherwise all experiments are the same.
+
+def rhs_fn(t, y, ydot):
+    """ rhs equations for the problem """
+    ydot[0] = y[1]
+    ydot[1] = -g
+
+def root_fn(t, y, out):
+    """ root function to check the object reached height Y1 """
+    #print('t = ', t, ', Y = ', y[0], ', v = ', y[1])
+    #print('Y1=', Y1, x)
+    out[0] = 10 - y[0]
+    return 0
+
+def onroot_va(t, y, solver):
+    """ function to handle interrupt cause by finding the root """
+    # Note: we know that flag can only 2 == Root found, so no need to check
+
+    # Teleport the object back to height Y0, but retain its speed
+    solver.reinit_IC(t, [Y0, y[1]])
+
+    return 0
+
+def onroot_vb(t, y, solver):
+    return 1
+
+def onroot_vc(t, y, solver):
+    if t > 28: # we have found 4 interruption points, so we stop
+        return 1
+
+    solver.reinit_IC(t, [Y0, y[1]])
+
+    return 0
+
+
+def print_results(experiment_no, result, require_no_roots=False):
+    ts, ys = result.values.t, result.values.y
+    # Print computed values at tspan
+    print('\n Experiment number: ', experiment_no)
+    print('--------------------------------------')
+    print('    t             Y               v')
+    print('--------------------------------------')
+
+    for (t, y, v) in zip(ts, ys[:, 0], ys[:, 1]):
+        print('%6.1f %15.4f %15.2f' % (t, y, v))
+
+
+    t_roots, y_roots = result.roots.t, result.roots.y
+    if not require_no_roots:
+        # Print interruption points
+        print('\n t_roots     y_roots        v_roots')
+        print('--------------------------------------')
+        if (t_roots is None) and (y_roots is None):
+            print('%6s %15s %15s' % (t_roots, y_roots, y_roots))
+        elif (t_roots is not None) and (y_roots is not None):
+            if np.isscalar(t_roots):
+                print('%6.1f %15.4f %15.2f' % (t_roots, y_roots[0], y_roots[1]))
+            else:
+                for (t, y, v) in zip(t_roots, y_roots[:, 0], y_roots[:, 1]):
+                    print('%6.1f %15.4f %15.2f' % (t, y, v))
+        else:
+            print('Error: one of (t_roots, y_roots) is None while the other not.')
+    else:
+        print('Computation failed.')
+
+# Set tspan to end at t_end1
+tspan = np.arange(0, t_end1 + 1, 1.0, np.float)
+#
+# 1. Solve the problem without interrupt function i.e. compute until
+#    the first root is found
+#
+solver = cvode.CVODE(rhs_fn, nr_rootfns=1, rootfn=root_fn, old_api=False)
```

Comment:
Why the raw cvode.CVODE function? Better as documentation to use ode('cvode', ...)